xamarin/Xamarin.Forms
# [iOS] Fix hard crash when horizontal GridViewLayout is refreshed

### Description of Change ###

Not sure if this is the right fix. When horizontal GridViewLayout is refreshed via a pull gesture, the application crashes. Previously, the code was trying to calculate missing cell count based on layout span. If you run the attached test, you'll see that `base.LayoutAttributesForElementsInRect(rect)` returns 9 items even though there are 8 visible items on the screen. The code calculated missing cell count to be 2 even though it should be 1. Consequently, we ended up with an index out of bounds error.

The fix allocates `UICollectionViewLayoutAttributes[]` based on the number of items in section 0, copies existing attributes from base, and appends new ones.

@hartez Do we need to change the Wikipedia image URLs?

@hartez @jfversluis @samhouts @jsuarezruiz Also, when the loading spinner is shown, the CollectionView seems to be resized, which forces it to show 3 rows of items instead of 4 before the next batch of items is added to the items source. Can you confirm if this is a bug or expected behavior? (Tested on XS)

### Issues Resolved ### 


- fixes #7593

### Testing Procedure ###
- Run the test
- Pull to refresh 4 times
- Verify that the application does not crash

### PR Checklist ###
